firmware/ec/test/suites/Test_ocmp_ltc4015.c

```diff
@@ -0,0 +1,1309 @@
+                        const void *data)
+{
+}
+#pragma GCC diagnostic pop
+/* ================================ Tests =================================== */
+/* Values are used in the below function are taken as per the datasheet*/
+void test_ocmp_ltc4015_probe(void)
+{
+    POSTData postData;
+    LTC4015_GpioConfig[0x05] = 0;
+
+    /* Test with the actual value */
+    LTC4015_regs[LTC4015_REG_SYSTEM_STATUS_INDICATOR] = LTC4015_CHARGER_ENABLED;
+    TEST_ASSERT_EQUAL(POST_DEV_FOUND, LTC4015_fxnTable.cb_probe(
+                                          &gbc_pwr_int_bat_charger, &postData));
+    TEST_ASSERT_EQUAL(OCGPIO_CFG_OUTPUT, LTC4015_GpioConfig[0x05]);
+
+    /* Test with an incorrect value */
+    LTC4015_regs[LTC4015_REG_SYSTEM_STATUS_INDICATOR] =
+        ~LTC4015_CHARGER_ENABLED;
+    TEST_ASSERT_EQUAL(
+        POST_DEV_MISSING,
+        LTC4015_fxnTable.cb_probe(&gbc_pwr_int_bat_charger, &postData));
+    /* Test with a missing device */
+    TEST_ASSERT_EQUAL(POST_DEV_MISSING, LTC4015_fxnTable.cb_probe(
+                                            &gbc_pwr_invalid_dev, &postData));
+
+    /* Test with a missing bus */
+    TEST_ASSERT_EQUAL(POST_DEV_MISSING, LTC4015_fxnTable.cb_probe(
+                                            &gbc_pwr_invalid_bus, &postData));
+}
+
+void test_ocmp_ltc4015_init(void)
+{
+    /* Now try to init with a pin associated */
+    AlertData alert_data = {
+        .subsystem = 1,
+        .componentId = 4,
+        .deviceId = 1,
+    };
+    AlertData *alert_data_cp = malloc(sizeof(AlertData));
+    *alert_data_cp = alert_data;
+    TEST_ASSERT_EQUAL(POST_DEV_CFG_DONE,
+                      LTC4015_fxnTable.cb_init(&gbc_pwr_int_bat_charger,
+                                               &fact_lithiumIon_cfg,
+                                               alert_data_cp));
+    TEST_ASSERT_EQUAL(LTC4015_INIT_BATTERY_VOLTAGE_LOW_LIMIT,
+                      LTC4015_regs[LTC4015_REG_BATTERY_VOLTAGE_LOW_LIMIT]);
+    TEST_ASSERT_EQUAL(LTC4015_INIT_BATTERY_VOLTAGE_HIGH_LIMIT,
+                      LTC4015_regs[LTC4015_REG_BATTERY_VOLTAGE_HIGH_LIMIT]);
+    TEST_ASSERT_EQUAL(LTC4015_INIT_CHARGE_CURRENT_LOW_LIMIT,
+                      LTC4015_regs[LTC4015_REG_CHARGE_CURRENT_LOW_LIMIT]);
+    TEST_ASSERT_EQUAL(LTC4015_INIT_INPUT_VOLTAGE_LOW_LIMIT,
+                      LTC4015_regs[LTC4015_REG_INPUT_VOLTAGE_LOW_LIMIT]);
+    TEST_ASSERT_EQUAL(LTC4015_INIT_INPUT_CURRENT_HIGH_LIMIT,
+                      LTC4015_regs[LTC4015_REG_INPUT_CURRENT_HIGH_LIMIT]);
+    TEST_ASSERT_EQUAL(LTC4015_INIT_INPUT_CURRENT_LIMIT_SETTING,
+                      LTC4015_regs[LTC4015_REG_INPUT_CURRENT_LIMIT_SETTING]);
+    TEST_ASSERT_EQUAL_HEX16(LTC4015_EVT_BVL | LTC4015_EVT_BVH |
+                                LTC4015_EVT_IVL | LTC4015_EVT_ICH |
+                                LTC4015_EVT_BCL,
+                            LTC4015_regs[LTC4015_REG_ENABLE_LIMIT_MONITIOR]);
```

Comment:
can we do assertion check for icharge , vcharge , enablechargerstatealerts and alert call back as well ?

Reply:
Assertion check for icharge , vcharge , enablechargerstatealerts is added and alert call back  is already taken care into alert function call.